### ISSUE TITLE
[AUR Package]

### ISSUE THREAD

**Issue by Twig6943** [ivan-hc/AM#959]
**Is your feature request related to a problem? Please describe.**
I want to add "AM/Appman" to my [rebos](https://docs.rs/crate/rebos/latest) config.

**Comment by Twig6943** [comment]
@ivan-hc very WIP PKGBUILD
Also does AM run on other CPU architectures? AFAIK it only runs on X86_64
```
pkgname=appmanager
pkgver=1.0.0 # Replace with the actual version
pkgrel=1
pkgdesc="AppImage Manager for Linux"
url="https://github.com/ivan-hc/AM"
license=('MIT')
arch=('x86_64')
depends=('glib2' 'libappimage')
makedepends=('git' 'cmake' 'curl')

# Function to get the latest version from GitHub releases
_get_latest_release() {
    curl --silent "https://api.github.com/repos/ivan-hc/AM/releases/latest" |
    grep '"tag_name":' |
    sed -E 's/.*"([^"]+)".*/\1/'
}

# Fetching the latest version
latest_version=$(_get_latest_release)
source=("https://github.com/ivan-hc/AM/releases/download/${latest_version}/AM-Installer")

# Generate the checksum dynamically
generate_sha256sums() {
    curl -L -o "AM-Installer" "$source" && sha256sum "AM-Installer" | awk '{print $1}'
}

build() {
    # Download the installer
    curl -L -o "AM-Installer" "$source"
    # Set the correct permissions
    chmod +x "AM-Installer"  # Make it executable

    # Generate and output the SHA256 checksum
    sha256sums=($(generate_sha256sums))
}

package() {
    install -Dm755 "AM-Installer" "$pkgdir/usr/bin/am-installer"
    # Create a symlink or move it as per your requirement
    ln -s /usr/bin/am-installer "$pkgdir/usr/bin/appmanager"
}
```
This downloads the latest am-installer script to /usr/bin I don't know how to make it execute the installer and then delete itself but its a start

**Comment by Twig6943** [comment]
> @ivan-hc very WIP PKGBUILD Also does AM run on other CPU architectures? AFAIK it only runs on X86_64
> 
> ```
> pkgname=appmanager
> pkgver=1.0.0 # Replace with the actual version
> pkgrel=1
> pkgdesc="AppImage Manager for Linux"
> url="https://github.com/ivan-hc/AM"
> license=('MIT')
> arch=('x86_64')
> depends=('glib2' 'libappimage')
> makedepends=('git' 'cmake' 'curl')
> 
> # Function to get the latest version from GitHub releases
> _get_latest_release() {
>     curl --silent "https://api.github.com/repos/ivan-hc/AM/releases/latest" |
>     grep '"tag_name":' |
>     sed -E 's/.*"([^"]+)".*/\1/'
> }
> 
> # Fetching the latest version
> latest_version=$(_get_latest_release)
> source=("https://github.com/ivan-hc/AM/releases/download/${latest_version}/AM-Installer")
> 
> # Generate the checksum dynamically
> generate_sha256sums() {
>     curl -L -o "AM-Installer" "$source" && sha256sum "AM-Installer" | awk '{print $1}'
> }
> 
> build() {
>     # Download the installer
>     curl -L -o "AM-Installer" "$source"
>     # Set the correct permissions
>     chmod +x "AM-Installer"  # Make it executable
> 
>     # Generate and output the SHA256 checksum
>     sha256sums=($(generate_sha256sums))
> }
> 
> package() {
>     install -Dm755 "AM-Installer" "$pkgdir/usr/bin/am-installer"
>     # Create a symlink or move it as per your requirement
>     ln -s /usr/bin/am-installer "$pkgdir/usr/bin/appmanager"
> }
> ```
> 
> This downloads the latest am-installer script to /usr/bin I don't know how to make it execute the installer and then delete itself but its a start

Improved version:

```
# Maintainer: Your Name <your.email@example.com>
pkgname=appmanager
pkgver=1.0.0 # Replace with the actual version
pkgrel=1
pkgdesc="AppImage Manager for Linux"
url="https://github.com/ivan-hc/AM"
license=('MIT')
arch=('x86_64') # Specify the architecture
depends=('glib2' 'libappimage') # Add any other dependencies here
makedepends=('git' 'cmake' 'curl') # Include build dependencies if needed

# Function to get the latest version from GitHub releases
_get_latest_release() {
    curl --silent "https://github.com/ivan-hc/AM/releases" |
    grep -oP '(?<=/AM/releases/tag/)[^"]+' |
    head -n 1
}

# Fetching the latest version
latest_version=$(_get_latest_release)
source=("https://github.com/ivan-hc/AM/releases/download/${latest_version}/AM-Installer")

# Generate the checksum dynamically
sha256sums=('SKIP') # Skip integrity checks for the source file

build() {
    # Download the installer
    curl -L -o "AM-Installer" "$source"
    # Set the correct permissions
    chmod +x "AM-Installer"  # Make it executable

    # Generate and output the SHA256 checksum dynamically
    echo "Generating checksum..."
    local checksum=$(sha256sum "AM-Installer" | awk '{print $1}')
    echo "Checksum: $checksum"
}

package() {
    install -Dm755 "AM-Installer" "$pkgdir/usr/bin/am-installer"
    # Create a symlink or move it as per your requirement
    ln -s /usr/bin/am-installer "$pkgdir/usr/bin/appmanager"
}
```

**Comment by ivan-hc** [comment]
> Also does AM run on other CPU architectures? AFAIK it only runs on X86_64

yes, on any architecture... I have it on an old i386 laptop born for Windows XP and that runs Debian 32bits

> This downloads the latest am-installer script to /usr/bin I don't know how to make it execute the installer and then delete itself but its a start

"AM" is [APP-MANAGER](https://github.com/ivan-hc/AM/blob/main/APP-MANAGER), you need to install it to /usr/bin as "`am`" to install "AM" as a system package.

----------------------------------------

NOTE: "AM-INSTALLER" is a total different scrip!

"AM-INSTALLER" only an immutable helper to use to install "AM" or "AppMan", and from with "AM" is installed the classic way:
```
/usr/local/bin/am
```
as a **symlink** to
```
/opt/am/APP-MANAGER
```
plus the directory of the modules
```
/opt/am/modules
```

------------------------------------

So, to recap:
- classic installation of "AM" is `/opt/am/APP-MANAGER` with a symlink named `/usr/local/bin/am`, you are the owner of the content of /opt/am and you can update "AM" and modules without root permissions like this
- system installation via package manager is the script "APP-MANAGER" installed as `/usr/bin/am`, the directory "modules" is under `/usr/lib`, so all paths requiring root permissions... this means that the package manager (in this case "YAY/PARU") is the only one that can update `/usr/bin/am` and `/usr/lib/modules`. In this case, "AM" is allowed only to update the installed apps, nothing more.

**Comment by ivan-hc** [comment]
This is how "AM-INSTALLER" appear when you run it

![354839133-82b21979-e99d-4bee-b466-716bac1e7e45](https://github.com/user-attachments/assets/886effbe-dda5-42cb-9ba5-ab32982d6efb)

so it is not "AM" itself

**Comment by Twig6943** [comment]
needs to be "AM" ?

**Comment by Twig6943** [comment]
This is what it does
![image](https://github.com/user-attachments/assets/e39470e0-3612-49e6-af67-6a75c20170f1)

**Comment by ivan-hc** [comment]
> This is what it does ![image](https://private-user-images.githubusercontent.com/119701717/378214594-e39470e0-3612-49e6-af67-6a75c20170f1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk0NDc4MTYsIm5iZiI6MTcyOTQ0NzUxNiwicGF0aCI6Ii8xMTk3MDE3MTcvMzc4MjE0NTk0LWUzOTQ3MGUwLTM2MTItNDllNi1hZjY3LTZhNzVjMjAxNzBmMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyMFQxODA1MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMTc0ODM1YTZkM2VlNjE5YTc5Njg4YmY2OGM3MDA0YjM3NTg4Y2VjMjhmZjYyN2FiY2JkYTllNzkwYTc3ZDg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.d6MLUj6eUIWKJWB5j-DH3EsQ4AVgnOC62vVLwONJa9U)

it is just an helper, nothing more.

This is "AM"/"AppMan":
- https://github.com/ivan-hc/AM/blob/main/APP-MANAGER

the sctipt you are running in the screenshot is this instead:
- https://github.com/ivan-hc/AM/blob/main/AM-INSTALLER

what "AM-INSTALLER" is to help choosing between AM and AppMan, for a classic installation.

## Classic installation
### "AM":
```
/opt/am/APP-MANAGER    ==>   with symlink named /usr/local/bin/am
/opt/am/modules
```
### AppMan:
```
$HOME/.local/bin/appman    ==>  (this is "APP-MANAGER" but renamed as "appman"
/path/to/appman/modules     ==> you choose where to install the apps and download modules, at first start
```

-----------------------------------

Now, what you should try to do with a non-classic installation is for "AM", so you need the script "APP-MANAGER", not the "AM-INSTALLER".

## Non-classic installation, via package manager ("AM" only)
This is the structure we have decided for a package manager to install AM:
```
/usr/bin/am                      ==> this is thye scrip "APP-MANAGER" renamed as "am"
/usr/lib/am/modules
```
with this structure, running the command `am -u` will only update the installed apps (still in /opt). "AM" and modules are updated via YAY/PARU.

This is because I wrote AM to be updatable without root permissions, in /opt

Being this an installation in /usr/bin and /usr/lib, root permissions are required, so updating AM and modules is disabled if AM is in /usr/bin instead od /opt/am/APP-MANAGER symlinked into a $PATH.

----------------------

That said, your PKGBUILD now only allows to run the installer, this means that whoever will run it, will choose for a "Classic installation" or one between AM and AppMan, or both.

**Comment by Twig6943** [comment]
Update but still experimental PKGBUILD
```
# Maintainer: Your Name <your.email@example.com>
pkgname=am
pkgver=1.0.0 # Replace with the actual version
pkgrel=1
pkgdesc="AppImage Manager for Linux"
url="https://github.com/ivan-hc/AM"
license=('MIT')
arch=('x86_64') # Specify the architecture
depends=('glib2' 'libappimage') # Add any other dependencies here
makedepends=('git' 'cmake' 'curl') # Include build dependencies if needed

build() {
    # Download the installer
    curl -L -o "APP-MANAGER" "https://raw.githubusercontent.com/ivan-hc/AM/main/APP-MANAGER"
    # Set the correct permissions
    chmod +x "APP-MANAGER"

    # Create the modules directory
    mkdir -p "modules"

    # Download additional modules
    for module in database.am install.am management.am sandboxes.am template.am; do
        curl -L -o "modules/$module" "https://raw.githubusercontent.com/ivan-hc/AM/main/modules/$module"
    done
}

package() {
    # Install the main executable
    install -Dm755 "APP-MANAGER" "/usr/bin/am"

    # Install the modules
    install -Dm644 "modules/"* "/usr/lib/am/modules/"

    #Create symlink
    ln -s "/usr/bin/APP-MANAGER" "/usr/bin/am"
}
```

**Comment by Twig6943** [comment]
```

**Comment by Twig6943** [comment]
Much love to @Oglo12
(seems to work properly)
```
# Maintainer: Fatih Bakal fatihbakal7@protonmail.com
pkgname=am
pkgver=9.0.0
pkgrel=1
pkgdesc="AppImage package manager to install, update and manage ALL of them, system-wide or locally."
url="https://github.com/ivan-hc/AM"
arch=('any')
license=('GPL3')

depends=(
    'glib2'
    'libappimage'
    'coreutils'
    'curl'
    'grep'
    'less'
    'sed'
    'wget'
    'git'
    'jq'
)

source=("https://raw.githubusercontent.com/ivan-hc/AM/main/APP-MANAGER")
sha256sums=('SKIP')

package() {
    # Create the main directory for AM
    install -dm755 "$pkgdir/opt/am/modules"
    install -dm755 "$pkgdir/usr/local/bin"

    # Create the uninstall script
    cat << 'EOF' > "$pkgdir/opt/am/remove"
#!/bin/sh
set -e
if [ "$(id -u)" -ne 0 ]; then echo "Permission denied"; exit 1; fi
rm -f /usr/local/bin/am
rm -R -f /opt/am
EOF
    chmod a+x "$pkgdir/opt/am/remove"

    # Move the downloaded main script
    install -Dm755 "${srcdir}/APP-MANAGER" "$pkgdir/opt/am/APP-MANAGER"
    ln -s /opt/am/APP-MANAGER "$pkgdir/usr/local/bin/am"

    # Download all modules from the GitHub repository
    module_base_url="https://raw.githubusercontent.com/ivan-hc/AM/main/modules"
    modules=$(curl -s https://api.github.com/repos/ivan-hc/AM/contents/modules | jq -r '.[].name')

    for module_name in $modules; do
        echo " ◆ Downloading $module_name..."
        curl -L -o "$pkgdir/opt/am/modules/$module_name" "$module_base_url/$module_name"
        chmod a+x "$pkgdir/opt/am/modules/$module_name"
    done

    # Change ownership to the current user
    chown -R "$USER" "$pkgdir/opt/am" 2>/dev/null || true
}
```

**Comment by Samueru-sama** [comment]
```
    # Create the uninstall script
    cat << 'EOF' > "$pkgdir/opt/am/remove"
#!/bin/sh
set -e
if [ "$(id -u)" -ne 0 ]; then echo "Permission denied"; exit 1; fi
rm -f /usr/local/bin/am
rm -R -f /opt/am
EOF
    chmod a+x "$pkgdir/opt/am/remove"
 ```
 
 you can fix that indentation issue by using tab spaces and adding a `<<-` to the heredoc. 
 
 
 ```
     # Create the uninstall script
	cat <<- 'EOF' > "$pkgdir/opt/am/remove"
	#!/bin/sh
	set -e
	if [ "$(id -u)" -ne 0 ]; then echo "Permission denied"; exit 1; fi
	rm -f /usr/local/bin/am
	rm -R -f /opt/am
	EOF
    chmod a+x "$pkgdir/opt/am/remove"
 ```
 
 Though don't try to copy it from the comment since github may replace the tab spaces here. 
 
 Also we don't depend on `libappimage` and I'm pretty sure we don't depend on `glib2` either

**Comment by Twig6943** [comment]
@Samueru-sama iirc libappimage is needed in order to run appimages on arch linux and its derivatives

**Comment by Twig6943** [comment]
> ```
>     # Create the uninstall script
>     cat << 'EOF' > "$pkgdir/opt/am/remove"
> #!/bin/sh
> set -e
> if [ "$(id -u)" -ne 0 ]; then echo "Permission denied"; exit 1; fi
> rm -f /usr/local/bin/am
> rm -R -f /opt/am
> EOF
>     chmod a+x "$pkgdir/opt/am/remove"
> ```
> 
> you can fix that indentation issue by using tab spaces and adding a `<<-` to the heredoc.
> 
> ```
>     # Create the uninstall script
>    cat <<- 'EOF' > "$pkgdir/opt/am/remove"
>    #!/bin/sh
>    set -e
>    if [ "$(id -u)" -ne 0 ]; then echo "Permission denied"; exit 1; fi
>    rm -f /usr/local/bin/am
>    rm -R -f /opt/am
>    EOF
>    chmod a+x "$pkgdir/opt/am/remove"
> ```
> 
> Though don't try to copy it from the comment since github may replace the tab spaces here.
> 
> Also we don't depend on `libappimage` and I'm pretty sure we don't depend on `glib2` either

also what issue are you talking about the pkgbuild works fine at least on the machines I've tried it on

**Comment by Twig6943** [comment]
That being said submitting a pkg to the AUR is hard as shit soo if you're reading this and know how to submit a pkg to the AUR pls do so

**Comment by knowngecko** [comment]
> That being said submitting a pkg to the AUR is hard as shit soo if you're reading this and know how to submit a pkg to the AUR pls do so

submitting a PKG to the AUR isn't very hard, you just clone an empty repo with your ssh key that you put into your AUR account, commit the changes you want and push back
Consider watching: https://www.youtube.com/watch?v=rcUl-Io2ZBY for more information and a step by step guide - it's what I used my first time

**Comment by Twig6943** [comment]
> > That being said submitting a pkg to the AUR is hard as shit soo if you're reading this and know how to submit a pkg to the AUR pls do so
> 
> submitting a PKG to the AUR isn't very hard, you just clone an empty repo with your ssh key that you put into your AUR account, commit the changes you want and push back Consider watching: https://www.youtube.com/watch?v=rcUl-Io2ZBY for more information and a step by step guide - it's what I used my first time

https://aur.archlinux.org/packages/am

**Comment by Samueru-sama** [comment]
> @Samueru-sama iirc libappimage is needed in order to run appimages on arch linux and its derivatives

No you don't need it, I'm on artix linux and I don't have the libappimage package, all you need is `fuse2` and hopefully in the future just `fuse3` once the static runtime is used on all other appimages.

>also what issue are you talking about the pkgbuild works fine at least on the machines I've tried it on

It is an **indentation** issue, that is it doesn't affect how the script works but it is badly written.

**Comment by Twig6943** [comment]
> > @Samueru-sama iirc libappimage is needed in order to run appimages on arch linux and its derivatives
> 
> No you don't need it, I'm on artix linux and I don't have the libappimage package, all you need is `fuse2` and hopefully in the future just `fuse3` once the static runtime is used on all other appimages.
> 
> > also what issue are you talking about the pkgbuild works fine at least on the machines I've tried it on
> 
> It is an **indentation** issue, that is it doesn't affect how the script works but it is badly written.

fixed

**Comment by Samueru-sama** [comment]
> fixed

you don't need to list both `fuse2` and `fuse3` 😅 with just `fuse2` it is enough. 

there are two types of appimages: 

* The old type2 appimages that have a dependency to `libfuse2`, which is provided by the `fuse2` package.

* And the type2-static appimages that don't have a dependency to `libfuse2` and just need a `fusermount` or `fusermount3` binary, which you get with either the `fuse2` or `fuse3`. (If you find this confusing go and tell to probono lmao).

Because most appimages still depend on `libfuse2` you only need to list `fuse2` as a dependency since it covers both, and hopefully in the future it can changed to fuse3 but not right now.

**Comment by Twig6943** [comment]
> > fixed
> 
> you don't need to list both `fuse2` and `fuse3` 😅 with just `fuse2` it is enough.
> 
> there are two types of appimages:
> 
> * The old type2 appimages that have a dependency to `libfuse2`, which is provided by the `fuse2` package.
> * And the type2-static appimages that don't have a dependency to `libfuse2` and just need a `fusermount` or `fusermount3` binary, which you get with either the `fuse2` or `fuse3` package. (If you find this confusing go and tell to probono lmao).
> 
> Because most appimages still depend on `libfuse2` you only need to list `fuse2` as a dependency since it covers both, and hopefully in the future it can changed to fuse3 but not right now.

going to mark both as a dependency (lazy to change later & its not like fuse3 takes up 25 gigs)
